File: generators/linear-generator.js
function LinearGenerator(a, b, m, x0) {
	this._a = a;
	this._b = b;
	this._m = m;
	this._current = x0;
}

LinearGenerator.prototype.next = function () {
	var current = this._current;
	this._current = (this._a * this._current + this._b) % this._m;
	return current;
}

module.exports = LinearGenerator;

File: generators/linear_generator.py
class LinearGenerator:
    def __init__(self, a, b, m, x0):
        self._a = a
        self._b = b
        self._m = m
        self._current = x0

    def next(self):
        current = self._current
        self._current = (self._a * self._current + self._b) % self._m

        return current
